src/asynk/async_worker.rs

```diff
@@ -1,79 +1,72 @@
-    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
-{
-    #[builder(setter(into))]
-    pub queue: AsyncQueue<Tls>,
-    #[builder(setter(into))]
-    pub task_type: Option<String>,
+#[derive(TypedBuilder)]
+pub struct AsyncWorker<'a> {
     #[builder(setter(into))]
+    pub queue: &'a mut dyn AsyncQueueable,
+    #[builder(default=DEFAULT_TASK_TYPE.to_string() , setter(into))]
+    pub task_type: String,
+    #[builder(default, setter(into))]
     pub sleep_params: SleepParams,
-    #[builder(setter(into))]
+    #[builder(default, setter(into))]
     pub retention_mode: RetentionMode,
 }
-impl<Tls> AsyncWorker<Tls>
-where
-    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
-    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
-    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
-    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
-{
-    pub async fn run(&mut self, task: Task) {
+impl<'a> AsyncWorker<'a> {
+    pub async fn run(&mut self, task: Task) -> Result<(), Error> {
         let result = self.execute_task(task).await;
         self.finalize_task(result).await
     }
     async fn execute_task(&mut self, task: Task) -> Result<Task, (Task, String)> {
         let actual_task: Box<dyn AsyncRunnable> =
             serde_json::from_value(task.metadata.clone()).unwrap();
 
-        let task_result = actual_task.run(&mut self.queue).await;
+        let task_result = actual_task.run(self.queue).await;
         match task_result {
             Ok(()) => Ok(task),
             Err(error) => Err((task, error.description)),
         }
     }
-    async fn finalize_task(&mut self, result: Result<Task, (Task, String)>) {
+    async fn finalize_task(&mut self, result: Result<Task, (Task, String)>) -> Result<(), Error> {
         match self.retention_mode {
-            RetentionMode::KeepAll => {
-                match result {
-                    Ok(task) => self
-                        .queue
+            RetentionMode::KeepAll => match result {
+                Ok(task) => {
+                    self.queue
                         .update_task_state(task, FangTaskState::Finished)
-                        .await
-                        .unwrap(),
-                    Err((task, error)) => self.queue.fail_task(task, &error).await.unwrap(),
-                };
-            }
-            RetentionMode::RemoveAll => {
-                match result {
-                    Ok(task) => self.queue.remove_task(task).await.unwrap(),
-                    Err((task, _error)) => self.queue.remove_task(task).await.unwrap(),
-                };
-            }
+                        .await?;
+                    Ok(())
+                }
+                Err((task, error)) => {
+                    self.queue.fail_task(task, &error).await?;
+                    Ok(())
+                }
+            },
+            RetentionMode::RemoveAll => match result {
```

Comment:
I think in this case, you don't care about the result, so it can be replaced with just
```
self.queue.remove_task(task).await?;
```